scanner/config/config.go

```diff
@@ -1,9 +1,143 @@
 package config
 
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/rs/zerolog"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	// DefaultConfiguration provides the default values for the scanner configuration.
+	DefaultConfiguration = Config{
+		HTTPListenAddr: ":9443",
+		GRPCListenAddr: ":8443",
+		Indexer: IndexerConfig{
+			Enable:       true,
+			DBConnString: "postgresql:///postgres?host=/var/run/postgresql",
+		},
+		Matcher: MatcherConfig{
+			Enable:       true,
+			DBConnString: "postgresql:///postgres?host=/var/run/postgresql",
+		},
+		// Default is empty.
+		MTLS: MTLSConfig{
+			CertsDir: "",
+		},
+		LogLevel: LogLevel(zerolog.InfoLevel),
+	}
+)
+
 // Config represents the Scanner configuration parameters.
 type Config struct {
-	// Mode specifies the mode in which Scanner will run.
-	//
-	// The options are: Indexer, Matcher, Combo.
-	Mode string `json:"mode"`
+	Indexer        IndexerConfig `yaml:"indexer"`
+	Matcher        MatcherConfig `yaml:"matcher"`
+	HTTPListenAddr string        `yaml:"http_listen_addr"`
+	GRPCListenAddr string        `yaml:"grpc_listen_addr"`
+	MTLS           MTLSConfig    `yaml:"mtls"`
```

Comment:
up to you, but I think just TLS is sufficient. Don't think we need to make it clear that we use mTLS

Reply:
There are files in that directory related to mTLS, so sing MTLS here makes it more obvious what everything inside the directory is.